library/src/commonTest/kotlin/com/ioki/passenger/api/MapApiErrorTest.kt

```diff
@@ -0,0 +1,105 @@
+import io.ktor.util.InternalAPI
+import io.ktor.utils.io.ByteReadChannel
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.test.expect
+
+@InternalAPI
+internal class MapApiErrorTest {
+
+    @Test
+    fun `mapApiError returns version outdated error`() = runTest {
+        val content = ByteReadChannel(
+            text = """
+                    {
+                      "api_errors": [
+                        { "message": "App version is outdated", "code": "version_outdated" }
+                      ]
+                    }
+                """,
+        )
+
+        val fakeHttpClient = createHttpClient(HttpStatusCode.NotAcceptable, content)
+        val fakeResponse = fakeHttpClient.get("https://127.0.0.1")
+
+        val fakeApiErrorInterceptor = object : ApiErrorInterceptor {
+            override fun intercept(apiErrors: List<ApiErrorBody.ApiError>, httpResponseCode: Int): Boolean {
+                val isAppVersionOutdated = apiErrors.any { it.code == "version_outdated" }
+                val isNotAcceptable = httpResponseCode == HttpStatusCode.NotAcceptable.value
+                return isAppVersionOutdated && isNotAcceptable
+            }
+        }
+
+        mapApiError(fakeResponse, listOf(fakeApiErrorInterceptor)).let { result ->
+            assertTrue { result is Result.Error.Api.Intercepted }
+            assertEquals(HttpStatusCode.NotAcceptable.value, result.httpStatusCode)
+        }
+    }
+
+    @Test
+    fun `mapApiError returns unauthorized error`() = runTest {
+        val content = ByteReadChannel.Empty
+
+        val fakeHttpClient = createHttpClient(HttpStatusCode.Unauthorized, content)
+        val fakeResponse = fakeHttpClient.get("https://127.0.0.1")
+
+        val fakeApiErrorInterceptor = object : ApiErrorInterceptor {
+            override fun intercept(apiErrors: List<ApiErrorBody.ApiError>, httpResponseCode: Int): Boolean {
+                return httpResponseCode == HttpStatusCode.Unauthorized.value
+            }
+        }
+
+        mapApiError(fakeResponse, listOf(fakeApiErrorInterceptor)).let { result ->
+            assertTrue { result is Result.Error.Api.Intercepted }
+            assertEquals(HttpStatusCode.Unauthorized.value, result.httpStatusCode)
+        }
+    }
+
+    @Test
+    fun `mapApiError returns generic error`() = runTest {
+        val content = ByteReadChannel.Empty
+
+        val fakeHttpClient = createHttpClient(HttpStatusCode.BadGateway, content)
+        val fakeResponse = fakeHttpClient.get("https://127.0.0.1")
+
+        val fakeApiErrorInterceptor = object : ApiErrorInterceptor {
+            override fun intercept(apiErrors: List<ApiErrorBody.ApiError>, httpResponseCode: Int): Boolean {
+                return httpResponseCode == HttpStatusCode.Unauthorized.value
+            }
+        }
+
+        mapApiError(fakeResponse, listOf(fakeApiErrorInterceptor)).let { result ->
```

Comment:
Don't use `let` here (and anywhere else 🙃 )
Testing should be divded into three sections:
`given, when, then`

We are testing `mapApiError`. This is part of `when`.
The result should be part of `then`.

Reply:
Good, forgot about that!

Reply:
https://github.com/ioki-mobility/kmp-passenger-api/pull/26/commits/fef79ddc2dccc190a8548730a88542b48a2a894b